lucene/core/src/java/org/apache/lucene/search/BoostQuery.java

```diff
@@ -104,6 +104,14 @@ public Query rewrite(IndexReader reader) throws IOException {
     return super.rewrite(reader);
   }
 
+  @Override
+  public void visit(QueryVisitor visitor) {
+    QueryVisitor v = visitor.getSubVisitor(BooleanClause.Occur.MUST, this);
+    if (v != null) {
```

Comment:
should you rather check for empty visitor here now? I guess we can assume that getSubVisitor never returns null?